normandy/recipes/filters.py

```diff
@@ -312,6 +312,47 @@ def capabilities(self):
         return set()
 
 
+class VersionRangeFilter(BaseFilter):
+    """
+    Match a user running a version in the given range. Uses a version compare
+    filter instead of simple string comparison like VersionFilter.
+
+    The version range is half-open, like Python ranges: If min is 72 and max
+    is 75, 72.0 will be include, 75.0 will not be. `min <= version < max`.
+
+    ..attribute:: type
+
+        ``version_range``
+
+    .. attribute:: min_version
+
+        :example: ``72.0b5``
+
+    .. attribute:: max_version
+
+        :example: ``75.0.1``
+    """
+
+    type = "version_range"
+    min_version = serializers.CharField()
+    max_version = serializers.CharField()
+
+    def to_jexl(self):
+        min_version = self.initial_data["min_version"]
+        max_version = self.initial_data["max_version"]
+
+        return "&&".join(
+            [
+                f'(env.version|versionCompare("{min_version}")>=0)',  # browser version >= min_version
+                f'(env.version|versionCompare("{max_version}")<0)',  # browser version < max_version
+            ]
+        )
+
+    @property
+    def capabilities(self):
+        return {"jexl.context.env.version", "jexl.transform.versionCompare"}
```

Comment:
So the idea is that we'll use `VersionRangeFilter` *instead* of `VersionFilter` for clients that support the capability, and fall back to `VersionFilter` for those that don't?

Reply:
Yes. Importantly, the plan I have here is that Normandy wouldn't actually be checking for that, and it would be up to the people writing recipes to manually pick the right one. It might be nice to have some explicit checks. I think that's mostly something we'd do in Normandy Devtools though.

I happen to know that Firefox 72 and above can safely use this, and that Firefox 71 and below cannot safely use this. Something like [bug 1563564](https://bugzilla.mozilla.org/show_bug.cgi?id=1563564) should make it easier to deal with that in tooling in the future.